metricq_wizard_backend/metricq/cluster_scanner.py

```diff
@@ -0,0 +1,521 @@
+            await self.create_issue_report(
+                scope_type="metric",
+                scope=metric,
+                severity="warning",
+                type="timeout",
+                source=metadata.get("source"),
+            )
+        except HistoryError as e:
+            await self.create_issue_report(
+                scope_type="metric",
+                scope=metric,
+                type="errored",
+                severity="info",
+                error=str(e),
+                source=metadata.get("source"),
+            )
+
+    async def check_metric_for_infinites(
+        self,
+        client: HistoryClient,
+        metric: str,
+        metadata: JsonDict,
+    ) -> None:
+        start_time = Timestamp(0)
+        end_time = Timestamp.from_now(Timedelta.from_string("7d"))
+
+        try:
+            result = await client.history_aggregate(
+                metric, start_time=start_time, end_time=end_time, timeout=60
+            )
+            if result.count and (
+                not math.isfinite(result.minimum) or not math.isfinite(
+                    result.maximum)
+            ):
+                await self.create_issue_report(
+                    scope_type="metric",
+                    scope=metric,
+                    type="infinite",
+                    severity="info",
+                    last_timestamp=str(result.timestamp.datetime.isoformat()),
+                    source=metadata.get("source"),
+                )
+            else:
+                await self.delete_issue_report(
+                    scope_type="metric",
+                    scope=metric,
+                    type="infinite",
+                )
+
+        except asyncio.TimeoutError:
+            # we should see this in the dead metrics check as well, so don't bother here.
+            pass
+        except HistoryError as e:
+            await self.create_issue_report(
+                scope_type="metric",
+                scope=metric,
+                type="errored",
+                severity="info",
+                error=str(e),
+                source=metadata.get("source"),
+            )
+
+    async def check_metric_name(self, metric: str):
+        assert self.db_metadata is not None
+
+        if not re.match(r"([a-zA-Z][a-zA-Z0-9_]+\.)+[a-zA-Z][a-zA-Z0-9_]+", metric):
+            doc = await self.db_metadata.get(metric)
+            await self.create_issue_report(
+                scope_type="metric",
+                scope=metric,
+                type="invalid_name",
+                severity="info",
+                source=doc.get("source"),
+            )
+        else:
+            await self.delete_issue_report(
+                scope_type="metric",
+                scope=metric,
+                type="invalid_name",
+            )
+
+    async def find_issues(self, currentPage, perPage, sortBy, sortDesc, **kwargs):
+        skip = (currentPage - 1) * perPage
```

Comment:
are you saying page is 1 indexed :imp: ?

Reply:
Yes, it comes straight from hell: https://bootstrap-vue.org/docs/components/pagination/